ofrak_core/test_ofrak/components/test_xattr_stub.py

```diff
@@ -0,0 +1,196 @@
+    value = xattr_stub_fixture.iterkeys()
+    assert "Function iterkeys" in caplog.text
+    assert type(value) == type(xattr_real_fixture.iterkeys())
+
+
+async def test_has_key(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.has_key(None)
+    assert "Function has_key" in caplog.text
+    assert type(value) == type(xattr_real_fixture.has_key("user.foo"))
+
+
+async def test_clear(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.clear()
+    assert "Function clear" in caplog.text
+    assert type(value) == type(xattr_real_fixture.clear())
+
+
+async def test_update(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.update(None)
+    assert "Function update" in caplog.text
+    assert type(value) == type(xattr_real_fixture.update({"user.foo": b"bar"}))
+
+
+async def test_copy(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.copy()
+    assert "Function copy" in caplog.text
+    assert type(value) == type(xattr_real_fixture.copy())
+
+
+async def test_setdefault(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.setdefault(None)
+    assert "Function setdefault" in caplog.text
+    assert type(value) == type(xattr_real_fixture.setdefault("user.foo"))
+
+
+async def test_keys(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.keys()
+    assert "Function keys" in caplog.text
+    assert type(value) == type(xattr_real_fixture.keys())
+
+
+async def test_itervalues(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.itervalues()
+    assert type(value) == type(xattr_real_fixture.itervalues())
+    next(value)
+    assert "Function itervalues" in caplog.text
+
+
+async def test_values(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.values()
+    assert "Function values" in caplog.text
+    assert type(value) == type(xattr_real_fixture.values())
+
+
+async def test_iteritems(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.iteritems()
+    assert type(value) == type(xattr_real_fixture.iteritems())
+    next(value)
+    assert "Function iteritems" in caplog.text
+
+
+async def test_items(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.items()
+    assert "Function items" in caplog.text
+    assert type(value) == type(xattr_real_fixture.items())
+
+
+async def test_listxattr(caplog):
+    value = xattr_stub.listxattr(None)
+    assert "Function listxattr" in caplog.text
+    assert type(value) == type(xattr.listxattr(os.path.join(EXAMPLE_DIRECTORY, "README.md")))
+
+
+async def test_getxattr(caplog):
+    value = xattr_stub.getxattr(None, None)
+    assert "Function getxattr" in caplog.text
+    assert type(value) == type(
+        xattr.getxattr(os.path.join(EXAMPLE_DIRECTORY, "README.md"), "user.foo")
```

Comment:
This assumes that the xattr "user.foo" is already set. What happens if it isn't set?